com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -117,22 +117,36 @@ public static BrainParametersProto ToProto(this BrainParameters bp, string name,
         /// <param name="isTraining">Whether or not the Brain is training.</param>
         public static BrainParametersProto ToBrainParametersProto(this ActionSpec actionSpec, string name, bool isTraining)
         {
-            actionSpec.CheckNotHybrid();
-
             var brainParametersProto = new BrainParametersProto
             {
                 BrainName = name,
                 IsTraining = isTraining
             };
-            if (actionSpec.NumContinuousActions > 0)
+            var actionSpecProto = new ActionSpecProto
+            {
+                NumContinuousActions = actionSpec.NumContinuousActions,
+                NumDiscreteActions = actionSpec.NumDiscreteActions,
+            };
+            if (actionSpec.BranchSizes != null)
             {
-                brainParametersProto.VectorActionSize.Add(actionSpec.NumContinuousActions);
-                brainParametersProto.VectorActionSpaceType = SpaceTypeProto.Continuous;
+                actionSpecProto.DiscreteBranchSizes.AddRange(actionSpec.BranchSizes);
             }
-            else if (actionSpec.NumDiscreteActions > 0)
+            brainParametersProto.ActionSpec = actionSpecProto;
+
+            var supportHybrid = Academy.Instance.TrainerCapabilities == null || Academy.Instance.TrainerCapabilities.HybridActions;
```

Comment:
we support hybrid actions if the TrainerCapabilities is no?

Reply:
or is it supposed to be `!= &&`

Reply:
TrainerCapabilities is null if there's no trainer connected. So in that case, we support whatever code is currently running in the package.